src/main/groovy/life/qbic/utils/DataSetParser.groovy

```diff
@@ -0,0 +1,35 @@
+package life.qbic.utils
+
+/**
+ * A parser that takes an input data structure and tries to convert it to a certain type dataset
+ * of type T.
+ *
+ * The user of this interface can expect a valid dataset of type T if no exceptions have
+ * been thrown after parsing of data structure has been called.
+ *
+ * @since 1.7.0
+ */
+interface DataSetParser<T> {
+
+    /**
+     * Tries to parse and validate a data structure from a given path on the filesystem.
+     *
+     * The root path must be absolute!
+     *
+     * Given a dataset with the structure:
+     *
+     * - /SomePath/MyDataset
+     *      |- myFile.txt
+     *      |- anotherFile.txt
+     *
+     * Then /SomePath/MyDataset would be the root path.
+     *
+     * @param rootPath The root path of the dataset structure, represents the top level of the
+     * hierarchical data set structure.
+     * @return A successfully parsed and validated dataset of type T
```

Comment:
```suggestion
     * @return A successfully parsed and validated dataset
```
Spelling out the type (can be anything) here is unnecessary and can lead to confusion. The return type is automatically filled in in the JavaDoc due to the method signature not being `def`